crates/miden-tx/src/errors/mod.rs

```diff
@@ -40,6 +42,18 @@ pub enum NoteCheckerError {
     },
 }
 
+impl NoteCheckerError {
+    pub fn into_transaction_executor_error(self) -> Option<TransactionExecutorError> {
```

Comment:
I didn't want to implement this as TryFrom because I didn't want to introduce an error or anyhow error.

Ideally would implement this as a From but I need to update the NoteCheckerError if possible.